src/langpacks.py

```diff
@@ -0,0 +1,332 @@
+# Copyright 2025 Canonical
+# See LICENSE file for licensing details.
+
+"""Representation of the langpacks service."""
+
+import logging
+import os
+import shutil
+from pathlib import Path
+from subprocess import PIPE, STDOUT, CalledProcessError, run
+
+import charms.operator_libs_linux.v0.apt as apt
+from charms.operator_libs_linux.v0.apt import PackageError, PackageNotFoundError
+from launchpadlib.launchpad import Launchpad
+
+logger = logging.getLogger(__name__)
+
+# Packages installed as part of the update process.
+PACKAGES = [
+    "build-essential",
+    "libgettextpo-dev",
+    "debhelper",
+    "fakeroot",
+    "python3-launchpadlib",
+    "python3-apt",
+    "dput",
+    "git",
+    "devscripts",
+    "lintian",
+]
+
+HOME = Path("~ubuntu").expanduser()
+REPO_LOCATION = HOME / "langpack-o-matic"
+
+
+class Langpacks:
+    """Represent a langpacks instance in the workload."""
+
+    def __init__(self):
+        logger.debug("Langpacks class init")
+
+    def setup_crontab(self):
+        """Configure the crontab for the service."""
+        try:
+            run(
+                [
+                    "su",
+                    "-c",
+                    "crontab src/crontab",
+                    "ubuntu",
+                ],
+                check=True,
+                stdout=PIPE,
+                stderr=STDOUT,
+                text=True,
+            )
+            logger.debug("Crontab configured.")
+            return
+        except CalledProcessError as e:
+            logger.debug(f"Installation of the crontab failed: {e.stdout}")
```

Comment:
```suggestion
            logger.debug(
            "Installation of the crontab failed: '%s'", e.stdout)
```

In general we try to avoid f-strings in logger calls

Reply:
(Applies elsewhere)